File: healthcheck.py
import requests
import time
from datetime import datetime

url = 'http://localhost:3000/healthcheck'

while True:
    current_time = datetime.now().time()
    if current_time >= time(6, 0, 0) and current_time <= time(20, 0, 0):
        response = requests.get(url)
        if response.status_code == 200:
            print('API is available')
        else:
            print('API is unavailable')
    time.sleep(180) # wait for 3 minutes before checking the time again

File: Deployement/helpers/healthcheck.js
// This is a helper script to prevent render scaling down the dyno between 6am and 8pm.
// start command: node healthcheck.js
const http = require('http')

const url = 'http://localhost:3000/healthcheck'
//https://linky-backend.onrender.com

setInterval(() => {
  const currentTime = new Date().getTime();
  const startTime = new Date(currentTime).setHours(6, 0, 0, 0)
  const endTime = new Date(currentTime).setHours(20, 0, 0, 0)
  
  if (currentTime >= startTime && currentTime <= endTime) {
    http.get(url, (res) => {
      if (res.statusCode === 200 || res.statusCode === 201) {
        console.log('API is available')
      } else {
        console.log('API is unavailable')
      }
    })
  }
}, 2 * 60 * 1000);  // 2 minutes checking
